examples/hello/server.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "hello.h"
+
+/*
+ * this file represent a server receving a message from a client
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_pt_index_t pti;
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_le_t leh;
+	long num_messages;
+	char *p;
+	char msg[PTL_EV_STR_SIZE];
+	char data[50];
+	/* represent the memory used to receive the message */
+	ptl_le_t le = { .start = data,
+			.length = sizeof(data),
+			.ct_handle = PTL_CT_NONE,
+			.uid = PTL_UID_ANY,
+			.options = PTL_LE_OP_PUT };
+
+	if (argc < 2) {
+		num_messages = 0;
+		printf("Usage: client <number of possible received messages>\n default : 0\n");

```

Comment:
Why do we print the usage if we don't supplied an argument?

We could do that with a `-h`/`--help` but let's not go that route
We should document in the README & a comment at the top of `main` what are the expected arguments

For example:

```C
/*
 * this file represent a server receving a message from a client
 *
 * Usage: ./server [number of messages to wait]
 *
 * The server will exit after the specified number of messages have been received.
 * 0 messages means to wait forever, this is the default value.
 */
```

Reply:
(and typo: represent -> represents)